### PR TITLE
support creation of new vertices in LSCSR

### PR DESCRIPTION
**Pull request by meyerzinn**

### CODE REVIEW

**Comment by tewaro** [comment]
What happened to the vertices api? and what about vertex properties?

**Comment by meyerzinn** [comment]
> and what about vertex properties?

We already supported vertex properties. What do you mean by

> What happened to the vertices api?

File: libgalois/include/galois/graphs/LS_LC_CSR_Graph.h [hunk 1]
```diff
@@ -158,6 +158,27 @@ class LS_LC_CSR_Graph : private boost::noncopyable {
...
+  VertexTopologyID addVertexTopologyOnly() {
+    m_vertices.emplace_back();
+    if constexpr (HasVertexData) {
+      m_vertex_data.resize(m_vertices.size());
+    }
+    return m_vertices.size() - 1;
+  }
+
+  template <typename V = VertexData, typename = std::enable_if<HasVertexData>>
+  VertexTopologyID addVertices(std::vector<V> data) {
```
Comment (tewaro) [inline libgalois/include/galois/graphs/LS_LC_CSR_Graph.h:172]: Comment what the return represents.

File: libgalois/include/galois/graphs/LS_LC_CSR_Graph.h [hunk 2]
```diff
@@ -158,6 +158,27 @@ class LS_LC_CSR_Graph : private boost::noncopyable {
...
+    VertexTopologyID start = m_vertices.size();
+    m_vertices.resize(m_vertices.size() + data.size());
+    m_vertex_data.resize(m_vertices.size());
+
+    galois::do_all(galois::iterate(start, start + data.size()),
+                   [&](VertexTopologyID const& id) { getData(id) = data[id]; });
+    return start;
+  }
```
Comment (tewaro) [inline libgalois/include/galois/graphs/LS_LC_CSR_Graph.h]: Please add tests for this.

**Review by tewaro — APPROVED** [review]

File: libgalois/include/galois/graphs/LS_LC_CSR_Graph.h (hunk 2 above)
Comment (meyerzinn) [reply libgalois/include/galois/graphs/LS_LC_CSR_Graph.h]: lol valid point it was incorrect, fixed

**Pull request merged**